pkg/cassandra/encryption.go

```diff
@@ -0,0 +1,217 @@
+	if template.PodTemplateSpec.Spec.Containers == nil {
+		template.PodTemplateSpec.Spec.Containers = make([]corev1.Container, 0)
+	}
+
+	cassandraContainer := &corev1.Container{
+		Name: "cassandra",
+	}
+
+	cassandraContainerIdx, foundCassandra := FindContainer(template.PodTemplateSpec, "cassandra")
+	if foundCassandra {
+		cassandraContainer = template.PodTemplateSpec.Spec.Containers[cassandraContainerIdx].DeepCopy()
+	}
+
+	AddOrUpdateVolumeMount(cassandraContainer, &keystoreVolume, StoreMountFullPath(storeType, "keystore"))
+	AddOrUpdateVolumeMount(cassandraContainer, &truststoreVolume, StoreMountFullPath(storeType, "truststore"))
+
+	// If the container is not found, add it to the pod spec
+	// If the container is found, update it
+	if foundCassandra {
+		template.PodTemplateSpec.Spec.Containers[cassandraContainerIdx] = *cassandraContainer
+	} else {
+		template.PodTemplateSpec.Spec.Containers = append(template.PodTemplateSpec.Spec.Containers, *cassandraContainer)
+	}
+}
+
+func checkMandatoryEncryptionFields(encryptionStores *encryption.EncryptionStores) error {
+	if encryptionStores == nil {
+		return fmt.Errorf("EncryptionStores is required to set up encryption")
+	}
+	if encryptionStores.KeystoreSecretRef.Name == "" {
+		return fmt.Errorf("keystore secret ref was not set")
+	}
+	if encryptionStores.KeystorePasswordSecretRef.Name == "" {
+		return fmt.Errorf("keystore password secret ref was not set")
+	}
+	if encryptionStores.TruststoreSecretRef.Name == "" {
+		return fmt.Errorf("truststore secret ref was not set")
+	}
+	if encryptionStores.TruststorePasswordSecretRef.Name == "" {
+		return fmt.Errorf("truststore password secret ref was not set")
+	}
+	return nil
+}
+
+func StoreMountFullPath(storeType string, storeName string) string {
+	return fmt.Sprintf("%s/%s-%s", StoresMountPath, storeType, storeName)
+}
+
+func ClientEncryptionEnabled(template *DatacenterConfig) bool {
+	return template.CassandraConfig.CassandraYaml.ClientEncryptionOptions != nil && template.CassandraConfig.CassandraYaml.ClientEncryptionOptions.Enabled
+}
+
+func ServerEncryptionEnabled(template *DatacenterConfig) bool {
+	return template.CassandraConfig.CassandraYaml.ServerEncryptionOptions != nil && template.CassandraConfig.CassandraYaml.ServerEncryptionOptions.Enabled != nil && *template.CassandraConfig.CassandraYaml.ServerEncryptionOptions.Enabled
+}
+
+type EncryptionStoresPasswords struct {
+	ClientKeystorePassword   string
+	ClientTruststorePassword string
+	ServerKeystorePassword   string
+	ServerTruststorePassword string
+}
+
+func ReadEncryptionStoresSecrets(ctx context.Context, klusterKey types.NamespacedName, template *DatacenterConfig, remoteClient client.Client, logger logr.Logger) (EncryptionStoresPasswords, error) {
+	encryptionStoresPasswords := EncryptionStoresPasswords{}
+	if ClientEncryptionEnabled(template) {
+		if template.ClientEncryptionStores == nil || template.ClientEncryptionStores.KeystorePasswordSecretRef.Name == "" {
```

Comment:
Why not call `checkMandatoryEncryptionFields` here?

Reply:
done